x-pack/platform/plugins/shared/onechat/public/application/components/tools/form/sections/configuration_fields/workflow_configuration_fields.tsx

```diff
@@ -6,24 +6,54 @@
  */
 
 import React from 'react';
-import { EuiFormRow } from '@elastic/eui';
-import { useFormContext } from 'react-hook-form';
+import { EuiFormRow, EuiSelect } from '@elastic/eui';
+import { useFormContext, Controller } from 'react-hook-form';
 import { WorkflowPicker } from '../../components/workflow/workflow_picker';
 import type { WorkflowToolFormData } from '../../types/tool_form_types';
 import { i18nMessages } from '../../i18n';
 
 export const WorkflowConfiguration = () => {
   const {
+    control,
     formState: { errors },
   } = useFormContext<WorkflowToolFormData>();
 
   return (
-    <EuiFormRow
-      label={i18nMessages.configuration.form.workflow.workflowLabel}
-      isInvalid={!!errors.workflow_id}
-      error={errors.workflow_id?.message}
-    >
-      <WorkflowPicker />
-    </EuiFormRow>
+    <>
+      <EuiFormRow
+        label={i18nMessages.configuration.form.workflow.workflowLabel}
+        isInvalid={!!errors.workflow_id}
+        error={errors.workflow_id?.message}
+      >
+        <WorkflowPicker />
+      </EuiFormRow>
+      <EuiFormRow
+        label={i18nMessages.configuration.form.workflow.waitForCompletionLabel}
+        helpText={i18nMessages.configuration.form.workflow.waitForCompletionHelpText}
+        isInvalid={!!errors.wait_for_completion}
+        error={errors.wait_for_completion?.message}
+      >
+        <Controller
+          name="wait_for_completion"
+          control={control}
+          render={({ field: { ref, ...field } }) => (
+            <EuiSelect
+              options={[
+                {
+                  value: 'true',
+                  text: i18nMessages.configuration.form.workflow.waitForCompletionYesLabel,
+                },
+                {
+                  value: 'false',
+                  text: i18nMessages.configuration.form.workflow.waitForCompletionNoLabel,
+                },
```

Comment:
2000 called, they'd like their string-to-boolean workaround back. But EuiSelect do not accept boolean values, and I think we do want a select here (one label per option more explicit than a global label + yes/no), so 🤷. If there's a better way to do it, please tell.

Reply:
Yeah I started by doing that, and I felt like it was a worse user experience than a select with explicit labels for each option. Now if we're fine with the checkbox, I'm happy to revert.